### PR TITLE
test(cdc-400gb-3d): New cdc job 3d 400gb dataset

### PR DESCRIPTION
**Pull request by aleksbykov**
Added new job for testing cdc feature which is ready for GA.
New job will be run for 3d and operate with dataset size 400GB

## PR pre-checks (self review)

- [ ] ~I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)~
- [ ] ~I didn't leave commented-out/debugging code~
- [x] I added the relevant `backport` labels
- [ ] ~New configuration option are added and documented (in `sdcm/sct_config.py`)~
- [ ] ~I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)~
- [ ] ~All new and existing unit tests passed (CI)~
- [ ] ~I have updated the Readme/doc folder accordingly (if needed)~

### CODE REVIEW

File: data_dir/cdc_profile_400gb.yaml [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+  - name: nums_set
+    size: fixed(5)
+    population: exp(1..5)
+
+  - name: names_set
+    size: fixed(5)
+    population: exp(1..5)
+
+  - name: class_list
+    size: uniform(1..15)
```
Comment (amoskong) [inline data_dir/cdc_profile_400gb.yaml:61]: The size, population have different meaning for set and list.

- items number of collection
- size of item

When I used list in old cdc_profile.yaml of longevity 4h, I can see some 40M column, the workloads fail for that.

File: test-cases/longevity/longevity-cdc-3d-400gb.yaml [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+n_monitor_nodes: 1
+
+instance_type_db: 'i3.4xlarge'
+
+nemesis_class_name: 'ChaosMonkey'
+nemesis_interval: 5
+nemesis_during_prepare: false
+
+user_prefix: 'longevity-cdc-3d-400gb'
+space_node_threshold: 322122547200
```
Comment (amoskong) [inline test-cases/longevity/longevity-cdc-3d-400gb.yaml:26]: How do you calculate this number?

File: jenkins-pipelines/longevity-cdc-3d-400gb.jenkinsfile [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+// trick from https://github.com/jenkinsci/workflow-cps-global-lib-plugin/pull/43
+def lib = library identifier: 'sct@snapshot', retriever: legacySCM(scm)
+
+longevityPipeline(
+    backend: 'aws',
+    aws_region: 'eu-west-1',
+    test_name: 'longevity_test.LongevityTest.test_custom_time',
+    test_config: 'test-cases/longevity/longevity-cdc-3d-400gb.yaml',
+
+    timeout: [time: 360, unit: 'MINUTES']
```
Comment (amoskong) [inline jenkins-pipelines/longevity-cdc-3d-400gb.jenkinsfile:12]: The timeout is wrong.

File: data_dir/cdc_profile_400gb.yaml (hunk 1 above)
Comment (amoskong) [reply data_dir/cdc_profile_400gb.yaml:61]: reference:

- https://github.com/scylladb/scylla/issues/5904#issuecomment-595083868
- https://github.com/scylladb/scylla-cluster-tests/pull/1910

File: jenkins-pipelines/longevity-cdc-3d-400gb.jenkinsfile (hunk 3 above)
Comment (aleksbykov) [reply jenkins-pipelines/longevity-cdc-3d-400gb.jenkinsfile:12]: yes, i will change, once the stabilize dataset size

File: test-cases/longevity/longevity-cdc-3d-400gb.yaml (hunk 2 above)
Comment (aleksbykov) [reply test-cases/longevity/longevity-cdc-3d-400gb.yaml:26]: 300 * 1024^3

File: data_dir/cdc_profile_400gb.yaml (hunk 1 above)
Comment (aleksbykov) [reply data_dir/cdc_profile_400gb.yaml:61]: will also remove list from profile

**Review by amoskong — APPROVED** [review]

**Review by kbr- — APPROVED** [review]

File: test-cases/longevity/longevity-cdc-3d-400gb.yaml (hunk 2 above)
Comment (roydahan) [reply test-cases/longevity/longevity-cdc-3d-400gb.yaml:28]: Why do you want the nemesis to start only after 300gb?

File: test-cases/longevity/longevity-cdc-3d-400gb.yaml [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+experimental: false
```
Comment (roydahan) [inline test-cases/longevity/longevity-cdc-3d-400gb.yaml:29]: Let's remove this.

File: test-cases/longevity/longevity-cdc-3d-400gb.yaml [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+append_scylla_args: '--blocked-reactor-notify-ms 500 --abort-on-lsa-bad-alloc 1 --abort-on-seastar-bad-alloc --abort-on-internal-error 1 --abort-on-ebadf 1 --experimental-features cdc'
```
Comment (roydahan) [inline test-cases/longevity/longevity-cdc-3d-400gb.yaml:30]: Many of these are already in the defaults...

File: test-cases/longevity/longevity-cdc-3d-400gb.yaml [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+              "cassandra-stress user no-warmup profile=/tmp/cdc_profile_400gb_preimage_postimage.yaml ops'(insert=2,read1=1,update_number=1,update_all=2,update_name=1,delete1=1)' cl=QUORUM duration=4320m -port jmx=6868 -mode cql3 native -rate threads=100"
+             ]
+
+n_db_nodes: 6
+n_loaders: 4
+n_monitor_nodes: 1
+
+round_robin: true
+
+instance_type_db: 'i3.4xlarge'
```
Comment (roydahan) [inline test-cases/longevity/longevity-cdc-3d-400gb.yaml:21]: Does it have enough space?

**Comment by roydahan** [comment]
Any results from staging job?
Please add link here and run on 4.3.rc0.

**Comment by aleksbykov** [comment]
@roydahan 
I ran staging job for master, result was one issue were found.
I ran for 4.3 but it was limited with 6 hours. During this time no errors were detected.
I will run for 4.3.rc0 for 3days.

File: test-cases/longevity/longevity-cdc-3d-400gb.yaml (hunk 6 above)
Comment (aleksbykov) [reply test-cases/longevity/longevity-cdc-3d-400gb.yaml:21]: Monitor shows that total space is 4TB. I think it should be enough. Will run for 3days.

File: test-cases/longevity/longevity-cdc-3d-400gb.yaml (hunk 5 above)
Comment (aleksbykov) [reply test-cases/longevity/longevity-cdc-3d-400gb.yaml:30]: I thing, they will be overwritten if i left only one parameter.

File: test-cases/longevity/longevity-cdc-3d-400gb.yaml (hunk 4 above)
Comment (aleksbykov) [reply test-cases/longevity/longevity-cdc-3d-400gb.yaml:29]: all 3 line?

**Comment by aleksbykov** [comment]
BYO job failed after 13h: https://jenkins.scylladb.com/view/staging/job/scylla-staging/job/abykov/job/BYO-abykov-staging/39
Investigating why.

File: test-cases/longevity/longevity-cdc-3d-400gb.yaml (hunk 4 above)
Comment (roydahan) [reply test-cases/longevity/longevity-cdc-3d-400gb.yaml:29]: just experimental

File: test-cases/longevity/longevity-cdc-3d-400gb.yaml (hunk 5 above)
Comment (roydahan) [reply test-cases/longevity/longevity-cdc-3d-400gb.yaml:30]: but you don't need anything.
CDC becomes GA now.

File: test-cases/longevity/longevity-cdc-3d-400gb.yaml (hunk 2 above)
Comment (aleksbykov) [reply test-cases/longevity/longevity-cdc-3d-400gb.yaml:28]: removed

File: test-cases/longevity/longevity-cdc-3d-400gb.yaml (hunk 4 above)
Comment (aleksbykov) [reply test-cases/longevity/longevity-cdc-3d-400gb.yaml:29]: done

File: test-cases/longevity/longevity-cdc-3d-400gb.yaml (hunk 5 above)
Comment (aleksbykov) [reply test-cases/longevity/longevity-cdc-3d-400gb.yaml:30]: done

**Review by roydahan — APPROVED** [review]
Looks good.
If the test fails due to scylla issues, we can merge and continue investigations and followup runs on the official branch.

**Comment by aleksbykov** [comment]
After second run again scylla issue https://github.com/scylladb/scylla/issues/7482 was reproduced after 11h.

**Pull request merged**